File: ICanBoogie/protected/all/modules/categories/lib/Category.php
<?php

namespace BlogMVC\Modules\Categories;

use ICanBoogie\Routing\ToSlug;

class Category extends \ICanBoogie\ActiveRecord implements ToSlug
{
	public $id;
	public $name;
	public $slug;
	public $post_count;

	public function __toString()
	{
		return $this->name;
	}

	public function to_slug()
	{
		return $this->slug;
	}
}

File: phoenix/web/models/category.ex
defmodule Blogmvc.Category do
  use Blogmvc.Web, :model

  schema "categories" do
    field :name, :string
    field :slug, :string
    field :post_count, :integer
  end

  @doc """
  Builds a changeset based on the `struct` and `params`.
  """
  def changeset(struct, params \\ %{}) do
    struct
    |> cast(params, [:name, :slug, :post_count])
    |> validate_required([:name, :slug, :post_count])
  end
end
